src/witness_generator.ts

```diff
@@ -111,6 +112,72 @@ async function compileNativeBinary({
   shellExec(compileCmd);
 }
 
+function calculateSrcHashes(contents) {
+  const hashes = new Map<string, string>();
+  for (const entry of Array.from(contents.entries())) {
+    const checksum = crypto.createHash('md5').update(entry[1], 'utf8').digest('hex');
+    hashes.set(entry[0], checksum);
+  }
+  return hashes;
+}
+
+function loadOldSrcHashes(src) {
+  const hashesFile = path.join(src, '..', 'srcs.sum');
+  try {
```

Comment:
When the file does not exist return null directly. If some other unexpected error happens console log error and return null